rfcs/2022-04-20-12187-log-namespacing.md

```diff
@@ -0,0 +1,399 @@
+}
+```
+
+metadata
+
+```json
+{
+  "source_ip": "192.168.0.1",
+  "hostname": "localhost",
+  "source_type": "socket",
+  "ingest_timestamp": "2020-10-15T11:01:46.499555308Z"
+}
+```
+
+-----
+
+#### HTTP source / JSON codec / Vector namespace
+
+event
+
+```json
+{
+    "data": {
+      "mineral": "quartz",
+      "food": "sushi"
+    },
+    "path": "/foo/bar",
+    // headers and query params were previously placed directly on the root. This needs to be nested to avoid potential naming conflicts.
+    "headers": {
+      "Content-Type": "application/json"
+    },
+    "query_params": {
+      "page": 14,
+      "size": 3
+    }
+}
+```
+
+metadata
+
+```json
+{
+  "source_type": "http",
+  "ingest_timestamp": "2020-10-15T11:01:46.499555308Z"
+}
+```
+
+-----
+
+
+
+## Outstanding Questions / Alternatives
+
+### Minimize data on the event
+
+The split of which data belongs in the event vs event metadata can sometimes be ambiguous. An alternative is to store only the
+"data" in the event, and everything else ("metadata" and "vector metadata") in the event metadata.
+This will keep the event itself very clean and easy to access.
+
+If there is extra data needed for a sink (such as the key for kafka), then the user will either need to use VRL
+to move that into the event, or a special path syntax will need to be added to refer to data on metadata in a sink.
+
+### Don't use Event Metadata
+
+One of the reasons to move some data to the "event metadata" is to prevent "unnecessary" data from
+being written to a sink that is just metadata.
+
+Using event metadata is a significant amount of work, as well as cognitive burden for users to introduce
+a new concept.
+
+Vector has the concept of "semantic meaning" to assign explicit meaning to different parts of an event.
+Today all data on the event is generally considered to be the "data" when writing to a sink, and by
+storing data in the "event metadata", it is effectively preventing the sink from writing that by default.
+
+Instead, everything could be stored in the event (under appropriate namespacing to prevent key collisions).
+Semantic meaning can be used to determine where the actual "data" lives. By default, data that would have been
+stored in "event metadata" would be ignored by sinks, based on the semantic meaning automatically applied.
+
+This should be able to achieve the same thing as using "event metadata". The main downside is that it
+requires semantic meaning to be fully implemented and enabled, which is not done yet.
+
+### Strings as a root event field
+
+This RFC proposes allowing a string as the root event element in some cases. This has historically not been
+allowed, and many parts of the code will currently panic in such a situation, although it's certainly possible
+to allow this. The alternative is to always nest this under "data" instead.
+
+
+### Namespace names
+
+Right now the decoded "data" is placed under the prefix "data". Other names might make more sense, such as
```

Comment:
I'm partial to this idea of using whatever name is most appropriate for the source. I think users will find this more familiar.